### ISSUE TITLE
No automatic re-casing of entities

### ISSUE THREAD

**Issue by geoffreymcgill** [bridgedotnet/Bridge#2577]
As of Bridge 16.0 all entities will **not** be automatically re-cased by the Bridge compiler. 

The new default naming convention policy is **What-You-See-Is-What-You-Get** (WYSIWYG). If the member is configured as UpperCamelCase in C#, the Bridge compiler will create the member as UpperCamelCase in JavaScript.

In previous releases, Fields, Methods, and Properties were re-cased to **lowerCamelCamel** case in the generated JavaScript. For example, a method defined as `DoSomething()` in C# would be emitted as `doSomething()` in the JavaScript. Now the method will be available as `DoSomething()` in both C# and JavaScript.

Given the following C# sample, the JavaScript results below demonstrate the Before and After changes as of Bridge 16.0. This sample also includes the new `Bridge.define` [refinements](https://github.com/bridgedotnet/Bridge/issues/2561).

#### C#

```csharp
public class Program
{
    public static void Main()
    {
        var p = new Person();
        
        p.Name = "Sally";
        p.DoSomething();
    }
}

public class Person
{
    public string Name { get; set; }
    
    public void DoSomething() { }
}
```

#### JavaScript (Before)

```js
Bridge.define("Demo.Person", {
    config: {
        properties: {
            Name: null
        }
    },
    doSomething: function () {
    }
});

Bridge.define("Demo.Program", {
    $main: function () {
        var p = new Demo.Person();

        p.setName("Sally");
        p.doSomething();
    }
});
```

#### JavaScript (After)

```js
Bridge.define("Demo.Person", {
    properties: {
        Name: null
    },
    methods: {
        DoSomething: function () { }
    }
});

Bridge.define("Demo.Program", {
    main: function () {
        var p = new Demo.Person();

        p.Name = "Sally";
        p.DoSomething();
    }
});
```

The new `[Convention]` attribute (Issue #2477) ([docs](https://github.com/bridgedotnet/Bridge/wiki/Attribute-Reference#convention)) has been introduced to enable opt-in configuration of the entity notation (case) for Class, Properties, Methods, etc. 

Setting `[Convention(Notation.LowerCamelCase)]` on the **Person** class would result in the following JavaScript.

#### C# with Convention Attribute

```csharp
[Convention(Notation.LowerCamelCase)]
public class Person
{
    public string Name { get; set; }

    public void DoSomething() { }
}
```

#### JavaScript

```js
Bridge.define("Demo.Person", {
    properties: {
        name: null
    },
    methods: {
        doSomething: function () { }
    }
});

Bridge.define("Demo.Program", {
    main: function () {
        var p = new Demo.Person();

        p.name = "Sally";
        p.doSomething();
    }
});
```

## FAQ

**Q. How do I get the same naming convention (lowerCamelCase) used in previous releases?**
**A.** Apply the following `[assembly: Convention]` attribute to any .cs file within your project.

```csharp
[assembly: Convention(Target = ConventionTarget.Member, Notation = Notation.LowerCamelCase)]
```

Fine grain configuration is available by defining Properties, Methods, and Fields individually.

```csharp
[assembly: Convention(Member = ConventionMember.Property, Notation = Notation.LowerCamelCase)]
[assembly: Convention(Member = ConventionMember.Field, Notation = Notation.LowerCamelCase)]
[assembly: Convention(Member = ConventionMember.Method, Notation = Notation.LowerCamelCase)]
```

**Comment by TriEdgeAI** [comment]
<3

**Comment by geoffreymcgill** [comment]
Implemented and will be available with the Bridge 16.0 release.
